File: freeswitch/typescript/src/Utils/Speech/PercentSpeller.ts
import { Noun } from "./Noun";
import { NumberNoun } from "./NumberNoun";
import { NumberSpeller } from "./NumberSpeller";
import { Case } from "./Enums";
import { LexicalUnit } from "./LexicalUnit";

export class PercentSpeller {
    private static unit: Noun;

    static initialize() {
        this.unit = new Noun("percent", "sipivr.speech.percent");
        this.unit.Case = Case.Genitive;
    }

    static spell(amount: number): LexicalUnit[] {
        amount = NumberSpeller.round(amount, 2);
        let res: LexicalUnit[] = [];
        NumberSpeller.decomposeNumber(new NumberNoun(amount)).forEach(f => res.push(f));
        res.push(this.unit);

        return res;
    }
}

PercentSpeller.initialize();

File: sipivr.plugins/sipivr.speech/src/main/java/ru/sipivr/speech/core/PercentSpeller.java
package ru.sipivr.speech.core;


import ru.sipivr.speech.enums.Case;
import ru.sipivr.speech.enums.Currency;
import ru.sipivr.speech.enums.Gender;

import java.util.HashMap;

/**
 * Created by Admin on 24.01.2016.
 */
public class PercentSpeller {
    private static final Noun unit;

    private NumberSpeller numberSpeller;

    static {
        unit = new Noun("sipivr.speech.percent", "percent");
        unit.setCaseType(Case.Genitive);
    }

    public PercentSpeller(NumberSpeller numberSpeller) {
        this.numberSpeller = numberSpeller;
    }

    public CompositeLexicalUnit spell(double amount) {
        amount = numberSpeller.round(amount, 2);
        CompositeLexicalUnit res = new CompositeLexicalUnit();
        res.getChildren().add(numberSpeller.decomposeNumber(new NumberNoun(amount)));
        res.getChildren().add(unit);

        return res;
    }
}